### ISSUE TITLE
Incorrect column highlight position with non-English characters

### ISSUE THREAD

**Issue by tomasz1986** [vinsworldcom/nppColumnTools#7]
When using non-English characters, the highlight position gets moved further to the right. My guess is that the non-English characters are actually coded using multiple symbols.

For example, try to copy and paste the Korean word `닭` into Notepad++. This is just one syllable, but is comprised of 4 actual letters. The highlight position will move 3 spaces further to the right than necessary.

**Comment by vinsworldcom** [comment]
What version of Notepad++ are you using?  Do you have Settings => Preferences => MISC. => Use DirectWrite enabled?

**Comment by tomasz1986** [comment]
Notepad++ v7.8.9, which is the newest as of now, and I had DirectWrite disabled, but have just tried enabling it, and there is no difference. I have also checked a completely clean installation of Notepad++, but no difference either.

I can also say that this seems to happen with any non-Latin characters. Be it Korean, Polish, Czech - the position always seems to move right more than it should, as if 1 character was counted as 2.

**Comment by vinsworldcom** [comment]
Happy to accept a pull request that fixes this, but I only deal in Windows English (US) characters so have no experience with other character formats or how they are handled.  Sorry ..

**Comment by tomasz1986** [comment]
Yes, but please keep in mind that English also has words with non-Latin characters, e.g. `naïve`, `fiancé` or `coup d'état`. The problem affects all of these too.

I have tried with a few different monospace fonts, but there is no difference, regardless of the language. As soon as a non-Latin character appears in the line, the column highlight and ruler position gets incorrectly moved to the right.

**Comment by FaceCrap** [comment]
This is not limited to non-Latin characters. Since day 1 using this plugin, the column indicator in the ruler area has never lined up properly with the actual position in the editor area. 
Regardless of the rendering method, it is always misaligned.

Also, after disabling and enabling the ruler, there's always this error msg

SC_TECHNOLOGY_DEFAULT not detected.

Notepad++ 8.8.5 64-bit

**Comment by vinsworldcom** [comment]
@FaceCrap you should read the README:

> NOTE: The ruler does not line up correctly unless [SC_TECHNOLOGY_DEFAULT](https://www.scintilla.org/ScintillaDoc.html#SCI_GETTECHNOLOGY) is set.

and your pop-up message box is telling you that.

Go to Notepad++ preferences and set the rendering mode to GDI.

**Comment by FaceCrap** [comment]
And  what made you think I didn't already try that? Doesn't make any difference. It still doesn't line up. In fact, the same error pops op _with_ GDI mode.

<img width="318" height="243" alt="Image" src="https://github.com/user-attachments/assets/a816cb66-eead-4c3a-8063-ce1e37dc1ad6" />

**Comment by vinsworldcom** [comment]
Works for me:

<img width="1812" height="1087" alt="Image" src="https://github.com/user-attachments/assets/447d6611-5b78-4c09-b2a0-287b2c0eba28" />

maybe try setting the technology with a plugin like NppExec or PythonScript?

UPDATE:  Also, you need to restart Notepad++ after changing your rendering mode - it is not instantaneous.
